src/stores/app/service.go

```diff
@@ -1,55 +1,56 @@
 package app
 
 import (
+	"context"
 	"github.com/consensys/quorum-key-manager/pkg/app"
 	"github.com/consensys/quorum-key-manager/src/auth"
+	"github.com/consensys/quorum-key-manager/src/entities"
 	"github.com/consensys/quorum-key-manager/src/infra/log"
-	manifestsmanager "github.com/consensys/quorum-key-manager/src/infra/manifests/filesystem"
-	"github.com/consensys/quorum-key-manager/src/infra/postgres/client"
-	"github.com/consensys/quorum-key-manager/src/stores/api"
+	postgresinfra "github.com/consensys/quorum-key-manager/src/infra/postgres"
+	"github.com/consensys/quorum-key-manager/src/stores/api/http"
+	"github.com/consensys/quorum-key-manager/src/stores/api/manifest"
 	"github.com/consensys/quorum-key-manager/src/stores/connectors/stores"
 	"github.com/consensys/quorum-key-manager/src/stores/connectors/utils"
+	"github.com/consensys/quorum-key-manager/src/stores/connectors/vaults"
 	"github.com/consensys/quorum-key-manager/src/stores/database/postgres"
-	storesmanager "github.com/consensys/quorum-key-manager/src/stores/manager"
 )
 
-func RegisterService(a *app.App, logger log.Logger) error {
-	cfg := new(Config)
-	err := a.ServiceConfig(cfg)
-	if err != nil {
-		return err
-	}
-
-	// Create Postgres DB
-	postgresClient, err := client.New(cfg.Postgres)
-	if err != nil {
-		return err
-	}
+func RegisterService(
+	ctx context.Context,
+	a *app.App,
+	logger log.Logger,
+	postgresClient postgresinfra.Client,
+	manifests map[string][]entities.Manifest,
+	roles auth.Roles,
+) (*stores.Connector, error) {
+	// Data layer
 	db := postgres.New(logger, postgresClient)
 
-	// Create manifest reader
-	manifestReader, err := manifestsmanager.New(cfg.Manifest)
-	if err != nil {
-		return err
-	}
+	// Business layer
+	vaultsConnector := vaults.NewConnector(logger)
+	storesConnector := stores.NewConnector(roles, db, vaultsConnector, logger)
+	utilsConnector := utils.NewConnector(logger)
 
-	// Load auth manager service
-	authManager := new(auth.Manager)
-	err = a.Service(authManager)
-	if err != nil {
-		return err
+	// Service layer
+	router := a.Router()
+	http.NewStoresHandler(storesConnector).Register(router)
+	http.NewUtilsHandler(utilsConnector).Register(router)
+
+	manifestVaultHandler := manifest.NewVaultsHandler(vaultsConnector) // Manifest reading is synchronous, similar to a config file
+	for _, mnf := range manifests[entities.VaultKind] {
+		err := manifestVaultHandler.Register(ctx, mnf.Specs)
+		if err != nil {
+			return nil, err
+		}
 	}
 
-	// Create and register the stores service
-	storesConnector := stores.NewConnector(*authManager, db, logger)
-	utilsConnector := utils.NewConnector(logger)
-	api.New(storesConnector, utilsConnector).Register(a.Router())
-
-	service := storesmanager.New(storesConnector, manifestReader, db, logger)
-	err = a.RegisterService(service)
-	if err != nil {
-		return err
+	manifestStoreHandler := manifest.NewStoresHandler(storesConnector) // Manifest reading is synchronous, similar to a config file
```

Comment:
StoreHandler would inject to inject VaultHandler to grab the vault clients into the corresponding Stores, do I missing something?